SqlFundamentalsDemo/SqlDemo/sp_InsertEmployeeInfo.sql

```diff
@@ -0,0 +1,36 @@
+﻿CREATE PROCEDURE sp_InsertEmployeeInfo
+    @EmployeeName NVARCHAR(100) = NULL,
+    @FirstName NVARCHAR(50) = NULL,
+    @LastName NVARCHAR(50) = NULL,
+    @CompanyName NVARCHAR(20),
+    @Position NVARCHAR(30) = NULL,
+    @Street NVARCHAR(50),
+    @City NVARCHAR(20) = NULL,
+    @State NVARCHAR(50) = NULL,
+    @ZipCode NVARCHAR(50) = NULL
+AS
+BEGIN
+    SET NOCOUNT ON;
+    
+    IF ((@EmployeeName IS NULL OR LTRIM(RTRIM(@EmployeeName)) = '') AND 
+        (@FirstName IS NULL OR LTRIM(RTRIM(@FirstName)) = '') AND 
+        (@LastName IS NULL OR LTRIM(RTRIM(@LastName)) = ''))
+        THROW 51000, 'At least one of the fields (EmployeeName or FirstName or LastName) must be provided.', 1;
+         
+    IF (LEN(@CompanyName) > 20)
```

Comment:
Also a good practice to put values in variables, you can declare it as `DECLARE @MaxCompanyNameLength INT = 20` in the beginning after `SET NOCOUNT ON`